entity-service-impl/src/test/java/org/hypertrace/entity/query/service/converter/filter/NullFilteringExpressionConverterTest.java

```diff
@@ -0,0 +1,129 @@
+        .thenReturn(Optional.of("subDocPath1"));
+
+    IdentifierConverter identifierConverter = mock(IdentifierConverter.class);
+    when(identifierConverter.convert(any(), eq(REQUEST_CONTEXT)))
+        .thenReturn("attributes.subDocPath1");
+    when(identifierConverterFactory.getIdentifierConverter(
+            "column1", "subDocPath1", ValueType.STRING, REQUEST_CONTEXT))
+        .thenReturn(identifierConverter);
+
+    ConstantExpression constantExpression = ConstantExpression.of("null");
+    when(constantExpressionConverter.convert(constant, REQUEST_CONTEXT))
+        .thenReturn(constantExpression);
+
+    FilterTypeExpression filterTypeExpression =
+        nullFilteringExpressionConverter.convert(
+            columnIdentifier, Operator.EQ, constant, REQUEST_CONTEXT);
+    IdentifierExpression identifierExpression = IdentifierExpression.of("attributes.subDocPath1");
+
+    assertEquals(
+        LogicalExpression.builder()
+            .operator(LogicalOperator.OR)
+            .operands(
+                List.of(
+                    RelationalExpression.of(
+                        identifierExpression, RelationalOperator.NOT_EXISTS, constantExpression),
+                    RelationalExpression.of(
+                        identifierExpression, RelationalOperator.EQ, constantExpression)))
+            .build(),
+        filterTypeExpression);
+  }
+
+  @Test
+  void testNEqNull() throws ConversionException {
+    ColumnIdentifier columnIdentifier =
+        ColumnIdentifier.newBuilder().setColumnName("column1").build();
+    LiteralConstant constant =
+        LiteralConstant.newBuilder().setValue(Value.newBuilder().setString("null").build()).build();
+
+    when(entityAttributeMapping.getDocStorePathByAttributeId(REQUEST_CONTEXT, "column1"))
+        .thenReturn(Optional.of("subDocPath1"));
+
+    IdentifierConverter identifierConverter = mock(IdentifierConverter.class);
+    when(identifierConverter.convert(any(), eq(REQUEST_CONTEXT)))
+        .thenReturn("attributes.subDocPath1");
+    when(identifierConverterFactory.getIdentifierConverter(
+            "column1", "subDocPath1", ValueType.STRING, REQUEST_CONTEXT))
+        .thenReturn(identifierConverter);
+
+    ConstantExpression constantExpression = ConstantExpression.of("null");
+    when(constantExpressionConverter.convert(constant, REQUEST_CONTEXT))
+        .thenReturn(constantExpression);
+
+    FilterTypeExpression filterTypeExpression =
+        nullFilteringExpressionConverter.convert(
+            columnIdentifier, Operator.NEQ, constant, REQUEST_CONTEXT);
+    IdentifierExpression identifierExpression = IdentifierExpression.of("attributes.subDocPath1");
+
+    assertEquals(
+        LogicalExpression.builder()
+            .operator(LogicalOperator.OR)
+            .operands(
+                List.of(
+                    RelationalExpression.of(
+                        identifierExpression, RelationalOperator.EXISTS, constantExpression),
```

Comment:
> If it exists and is set to null, this would incorrectly return true

I was assuming that we never set it to null. But, I also don't see a use case, we doing a `NEQ null` filter

Reply:
I just realized we actually have a use case for `NEQ null` filter

Reply:
https://github.com/hypertrace/entity-service/pull/177